spec/support/fake_analytics_spec.rb

```diff
@@ -0,0 +1,455 @@
+require 'rails_helper'
+
+RSpec.describe FakeAnalytics do
+  subject(:analytics) { FakeAnalytics.new }
+
+  describe '#have_logged_event' do
+    context 'no arguments' do
+      let(:track_event) { -> { analytics.track_event :my_event } }
+      let(:code_under_test) { -> { expect(analytics).to have_logged_event } }
+
+      it 'raises if event was not logged' do
+        expect { code_under_test.call }.
```

Comment:
I just tested this locally and confirmed this works if we want pass the block/lambda directly:
```suggestion
        expect(&code_under_test).
```
(same for rest of file too)

Reply:
Ooh, that is nicer